src/rshim.c

```diff
@@ -2069,20 +2112,20 @@ rshim_backend_t *rshim_find_by_dev(void *dev)
   return NULL;
 }
 
+/*
+ * Must run with bd->mutex held.
+ * */
 int rshim_set_drop_mode(rshim_backend_t *bd, int value)
 {
   int old_value;
   int rt = 0;
 
-  pthread_mutex_lock(&bd->mutex);
   old_value = (int)bd->drop_mode;
   value = !!value;
   if (value == old_value) {
-    pthread_mutex_unlock(&bd->mutex);
     return -EALREADY;
   }
 
-  bd->drop_mode = 0;
```

Comment:
Any reason to remove this line? It could cause problem.

Reply:
added